packages/labs/signals/src/lib/watch.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {DirectiveResult, Part, directive} from 'lit/directive.js';
+import {AsyncDirective} from 'lit/async-directive.js';
+import {Signal} from 'signal-polyfill';
+import {SignalWatcher} from './signal-watcher.js';
+
+export class WatchDirective<T> extends AsyncDirective {
+  private __host?: SignalWatcher;
+
+  private __signal?: Signal.State<T> | Signal.Computed<T>;
+
+  private __watcher?: Signal.subtle.Watcher;
+
+  // We have to wrap the signal in a computed to work around a bug in the
+  // signal-polyfill: https://github.com/proposal-signals/signal-polyfill/issues/27
+  private __computed?: Signal.Computed<T | undefined>;
+
+  private __watch() {
+    if (this.__watcher === undefined) {
+      this.__computed = new Signal.Computed(() => {
+        return this.__signal?.get();
+      });
+      const watcher = (this.__watcher = new Signal.subtle.Watcher(() => {
+        this.__host?.updateWatchDirective(this as WatchDirective<unknown>);
+        watcher.watch();
+      }));
+      watcher.watch(this.__computed);
+    }
+  }
+
+  private __unwatch() {
```

Comment:
Should this remove the directive from the host's pending update list?

Reply:
It can cause an exception if we don't remove the directive. When a part is removed, its `parentNode` can become `null`. This could happen via some parent part, and then when this directive calls `setValue` it has no `parentNode` and this is invalid. 

```js
  html`${this.selected ? html`${watch(this.message}}` : ''}`

Reply:
done